src/ansys/fluent/core/utils/file_transfer_service.py

```diff
@@ -103,14 +147,43 @@ def download(self, file_name: str, local_file_name: Optional[str] = None):
         PyPIMConfigurationError
             If PyPIM is not configured.
         """
-        if not pypim.is_configured():
+        if not self.is_configured():
             raise PyPIMConfigurationError()
         elif self.file_service:
             if self.file_service.file_exist(file_name):
                 self.file_service.download_file(file_name, local_file_name)
             else:
                 raise FileNotFoundError("Remote file does not exist.")
 
+    def download_file(
+        self,
+        file_name: Union[list[str], str],
+        before_downloaded: Optional[Callable] = None,
+    ):
+        """Perform callback operation and
+        downloads a file if it's available to the server supported by
+        `PyPIM<https://pypim.docs.pyansys.com/version/stable/>`.
+
+        Parameters
+        ----------
+        file_name : str
+            File name
+        before_downloaded: Callable
+            Write a file.
+        """
+        files = [file_name] if isinstance(file_name, str) else file_name
+        for file in files:
+            if before_downloaded:
+                before_downloaded(
+                    os.path.basename(file) if self.is_configured() else file
+                )
+        if self.is_configured():
+            for file in files:
+                if os.path.isfile(file):
+                    print(f"\nFile already exists. File path:\n{file}\n")
+                else:
+                    self.download(os.path.basename(file), local_file_name=".")
+
 
 class RemoteFileHandler:
```

Comment:
as discussed, `RemoteFileHandler` will be removed (redundant). @hpohekar

Reply:
We will integrate this changes in a separate PR.